src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -629,6 +628,104 @@ void EIRPlantLoopHeatPump::calcLoadSideHeatTransfer(EnergyPlusData &state, Real6
     // currentLoad will be met and there should? be some adjustment based on outlet water temp limit?
 }
 
+void HeatPumpAirToWater::calcLoadSideHeatTransfer(EnergyPlusData &state, Real64 const availableCapacity, Real64 const currentLoad)
+{
+    Real64 CpLoad = this->loadSidePlantLoc.loop->glycol->getSpecificHeat(
+        state, state.dataLoopNodes->Node(this->loadSideNodes.inlet).Temp, "HeatPumpAirToWater::calcLoadSideHeatTransfer()");
+
+    // note loadSideHeatTransfer is non-negative, currentLoad is negative for cooling
+    this->loadSideHeatTransfer = min(availableCapacity, std::fabs(currentLoad));
+    auto minLoad = this->minimumPLR * availableCapacity;
+    if (fabs(currentLoad) < minLoad) {
+        this->loadSideHeatTransfer = minLoad;
+    }
+
+    // calculate load side outlet conditions
+    Real64 const loadMCp = this->loadSideMassFlowRate * CpLoad;
+    this->loadSideOutletTemp = this->calcLoadOutletTemp(this->loadSideInletTemp, this->loadSideHeatTransfer / loadMCp);
+}
+
+void HeatPumpAirToWater::calcPowerUsage(EnergyPlusData &state, Real64 availableCapacityBeforeMultiplier)
+{
+    // check to see how many heat pumps are needed
+    int numHeatPumpUsed = ceil(this->loadSideHeatTransfer / availableCapacityBeforeMultiplier);
+    Real64 currentLoadNthUnit = this->loadSideHeatTransfer - (numHeatPumpUsed - 1) * availableCapacityBeforeMultiplier;
+
+    //    calculate speed level
+    Real64 capacityLow = 0.0;
+    Real64 capacityHigh = 0.0;
+    Real64 loadSideOutletSetpointTemp = this->getLoadSideOutletSetPointTemp(state);
+    Real64 capacityModifierFuncTempLow = 1.0;
+    Real64 capacityModifierFuncTempHigh = 1.0;
+    // get speed level of the nth active heat pump
+    int speedLevel = 0;
+    for (int i = 0; i < this->numSpeeds; i++) {
+        capacityModifierFuncTempHigh =
+            Curve::CurveValue(state, this->capFuncTempCurveIndex[i], loadSideOutletSetpointTemp, this->sourceSideInletTemp);
+        capacityHigh = this->ratedCapacity[i] * capacityModifierFuncTempHigh;
+        speedLevel = i;
+        if (std::fabs(currentLoadNthUnit) <= capacityHigh) {
+            break;
+        } else {
+            capacityLow = capacityHigh;
+            capacityModifierFuncTempLow = capacityModifierFuncTempHigh;
+        }
+    }
+    // calculate power usage from EIR curves
+    Real64 eirModifierFuncTempLow = 1.0;
+    Real64 eirModifierFuncPLRLow = 1.0;
+    if (speedLevel > 0) {
```

Comment:
I noticed at the bottom of this function is `this->speedLevel += 1;` and that this conditional and the curve indexes below use a local variable `speedLevel`. This local variable is always = this->numSpeeds -1 here. This doesn't seem right.